lib/sinon/util/fake_xml_http_request.js

```diff
@@ -492,6 +492,12 @@ extend(FakeXMLHttpRequest.prototype, sinonEvent.EventTarget, {
         }
     },
 
+    setStatus: function setStatus(status) {
+        verifyRequestOpened(this);
+        this.status = typeof status === "number" ? status : 200;
+        this.statusText = FakeXMLHttpRequest.statusCodes[this.status];
```

Comment:
Can't do, the line above does the sanitizing, so `undefined` is treated as 200.

BTW, this is a piece of code moved over from `respond`.